mods/tnt/init.lua

```diff
@@ -232,6 +232,40 @@ end
 
 local function tnt_explode(pos, radius, ignore_protection, ignore_on_blast)
 	local pos = vector.round(pos)
+	-- scan for adjacent TNT nodes first, and enlarge the explosion
+	local vm1 = VoxelManip()
+	local p1 = vector.subtract(pos, 2)
+	local p2 = vector.add(pos, 2)
+	local minp, maxp = vm1:read_from_map(p1, p2)
+	local a = VoxelArea:new({MinEdge = minp, MaxEdge = maxp})
+	local data = vm1:get_data()
+	local count = 0
+	local c_tnt = minetest.get_content_id("tnt:tnt")
+	local c_tnt_burning = minetest.get_content_id("tnt:tnt_burning")
+	local c_tnt_boom = minetest.get_content_id("tnt:boom")
+	local c_air = minetest.get_content_id("air")
+
+	for z = pos.z - 2, pos.z + 2 do
+	for y = pos.y - 2, pos.y + 2 do
+	for x = pos.x - 2, pos.x + 2 do
+		local vi = a:index(x, y, z)
+		local cid = data[vi]
+		if cid == c_tnt or cid == c_tnt_boom or cid == c_tnt_burning then
+			count = count + 1
+			data[vi] = c_air
+		end
+	end
+	end
+	end
+
+	vm1:set_data(data)
+	vm1:write_to_map()
+	vm1:update_map()
```

Comment:
update_map() is a lighting update, so this can probably be removed and just have the 1 update later. This lighting update is a slow one so this will speed things up.